app/src/test/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModelTest.kt

```diff
@@ -135,8 +185,77 @@ class LoginSSOViewModelTest {
+
+        runTest { loginViewModel.establishSSOSession("", "") }
+        loginViewModel.loginState.loginSSOError shouldBeInstanceOf LoginError.DialogError.InvalidSSOCookie::class
+        coVerify(exactly = 0) { loginViewModel.registerClient(any()) }
+        coVerify(exactly = 0) { loginViewModel.navigateToConvScreen() }
+    }
+
+    @Test
+    fun `given HandleSSOResult is called, when ssoResult is null, then loginSSOError state should be none`() {
+        runTest { loginViewModel.handleSSOResult(null) }
+        loginViewModel.loginState.loginSSOError shouldBeEqualTo LoginError.None
+    }
+
+    @Test
+    fun `given HandleSSOResult is called, when ssoResult is failure, then loginSSOError state should be dialog error`() {
+        runTest { loginViewModel.handleSSOResult(DeepLinkResult.SSOLogin.Failure(SSOFailureCodes.Unknown)) }
+        loginViewModel.loginState.loginSSOError shouldBeEqualTo LoginError.DialogError.SSOResultError(SSOFailureCodes.Unknown)
+    }
+
+    @Test
+    fun `given HandleSSOResult is called, when SSOLoginResult is success, then establishSSOSession should be called once`() {
+        coEvery { getSSOLoginSessionUseCase.invoke(any(), any()) } returns SSOLoginSessionResult.Success(authSession)
+        coEvery { addAuthenticatedUserUseCase.invoke(any(), any()) } returns AddAuthenticatedUserUseCase.Result.Success(userId)
+        coEvery { registerClientUseCase.invoke(any(), any(), any()) } returns RegisterClientResult.Success(client)
+
+        runTest { loginViewModel.handleSSOResult(DeepLinkResult.SSOLogin.Success("", "")) }
+        coVerify(exactly = 1) { loginViewModel.navigateToConvScreen() }
     }
+
+
+    @Test
+    fun `given establishSSOSession is called, when addAuthenticatedUser returns UserAlreadyExists error, then UserAlreadyExists is passed`()
+    {
+        coEvery { getSSOLoginSessionUseCase.invoke(any(), any()) } returns SSOLoginSessionResult.Success(authSession)
+        coEvery { addAuthenticatedUserUseCase.invoke(any(), any()) } returns AddAuthenticatedUserUseCase.Result.Failure.UserAlreadyExists
+
+        runTest { loginViewModel.establishSSOSession("", "") }
+
+        loginViewModel.loginState.loginSSOError shouldBeInstanceOf LoginError.DialogError.UserAlreadyExists::class
+    }
+
+    @Test
+    fun `given establishSSOSession is called, when registerClientUseCase returns TooManyClients error, then TooManyClients is passed`() {
+        coEvery { getSSOLoginSessionUseCase.invoke(any(), any()) } returns SSOLoginSessionResult.Success(authSession)
+        coEvery { addAuthenticatedUserUseCase.invoke(any(), any()) } returns AddAuthenticatedUserUseCase.Result.Success(userId)
+        coEvery { registerClientUseCase.invoke(any(), any(), any()) } returns RegisterClientResult.Failure.TooManyClients
+
+        runTest { loginViewModel.establishSSOSession("", "") }
+
+        loginViewModel.loginState.loginSSOError shouldBeInstanceOf LoginError.TooManyDevicesError::class
+    }
```

Comment:
We need to assert that:

- getSSOLoginSessionUseCase is being invoked once
- addAuthenticatedUserUseCase is being invoked once
- registerClient is being invoked once